### DISCUSSION TITLE
background color for text box

### DISCUSSION THREAD

**Discussion by JohnSmall** [prawnpdf/prawn#1168] [General]
I see this question has been asked many times. And always closed without being resolved. 

So I'll ask again. How do you set the background color of a text_box?

If it's not totally simple e.g. `pdf.text_box, 'some text', background_color: 'BEBEBE'` then it should be made to be totally simple.

The documentation gives no clue. The closest mention is formatted_text_box, but that only allows you do set the color of the text, not the background color of the box.

**Comment by gettalong** [comment]
The prawn manual builder uses a helper method to create the colored boxes, see https://github.com/prawnpdf/prawn-manual_builder/blob/master/lib/prawn/manual_builder/example.rb#L328.

Also see page 55 of the Prawn manual on `text/formatted_callbacks.rb` where callback hooks are used to color a text box.

**Reply by PhilippePerret** [reply]
Unless I'm mistaken, this is a dead link.

**Reply by pointlessone** [reply]
Try one of previous tags (before 0.4.0) instead of `master` branch.

**Reply by pointlessone** [reply]
Or try [`colored_box`](https://github.com/prawnpdf/prawn-manual_builder/blob/4a7a4250ec37e2b725fd2fd3f1d6048735a1ebc9/lib/prawn/manual_builder/part.rb#L17) in 0.4.0.
